src/software.rs

```diff
@@ -0,0 +1,40 @@
+use std::time::Duration;
```

Comment:
I'm not loving the names... any ideas?

Reply:
I would probably go w/ `Role` or something.

Reply:
Updated to `Role` in the latest commit.